heron/ckptmgr/tests/java/com/twitter/heron/ckptmgr/CheckpointManagerServerTest.java

```diff
@@ -0,0 +1,387 @@
+
+    runServer();
+    // let the server warm up
+    Thread.sleep(2 * 1000);
+  }
+
+  @After
+  public void after() {
+    threadPools.shutdownNow();
+
+    checkpointManagerServer.stop();
+    checkpointManagerServer = null;
+
+    if (simpleCheckpointManagerClient != null) {
+      simpleCheckpointManagerClient.stop();
+
+      simpleCheckpointManagerClient.getNIOLooper().exitLoop();
+    }
+
+    serverLooper.exitLoop();
+    serverLooper = null;
+
+    threadPools = null;
+  }
+
+  @Test
+  public void testSaveInstanceState() throws Exception {
+    prepareClient(SimpleCheckpointManagerClient.RequestType.SAVE_INSTANCE_STATE);
+
+    runClient(simpleCheckpointManagerClient);
+    // let the client start and send request
+    Thread.sleep(2 * 1000);
+
+    verify(backendStorage).store(any(Checkpoint.class));
+    Assert.assertEquals(StatusCode.OK, simpleCheckpointManagerClient.getResponseStatus());
+    Message response = simpleCheckpointManagerClient.getResponse();
+    Assert.assertTrue(response instanceof CheckpointManager.SaveInstanceStateResponse);
+    Assert.assertEquals(CHECKPOINT_ID,
+        ((CheckpointManager.SaveInstanceStateResponse) response).getCheckpointId());
+    Assert.assertEquals(instance,
+        ((CheckpointManager.SaveInstanceStateResponse) response).getInstance());
+  }
+
+  @Test
+  public void testGetInstanceState() throws Exception {
+    doAnswer(new Answer<Checkpoint>() {
+      @Override
+      public Checkpoint answer(InvocationOnMock invocationOnMock) throws Throwable {
+        Checkpoint checkpoint = (Checkpoint) invocationOnMock.getArguments()[0];
+        checkpoint.setCheckpoint(saveInstanceStateRequest);
+        return checkpoint;
+      }
+    }).when(backendStorage).restore(any(Checkpoint.class));
+
+    prepareClient(SimpleCheckpointManagerClient.RequestType.GET_INSTANCE_STATE);
+
+    runClient(simpleCheckpointManagerClient);
+
+    // let the client start and send request
+    Thread.sleep(2 * 1000);
+
+    verify(backendStorage).restore(any(Checkpoint.class));
+    Assert.assertEquals(StatusCode.OK, simpleCheckpointManagerClient.getResponseStatus());
+
+    Message response = simpleCheckpointManagerClient.getResponse();
+    Assert.assertTrue(response instanceof CheckpointManager.GetInstanceStateResponse);
+    Assert.assertEquals(saveInstanceStateRequest.getCheckpoint(),
+        ((CheckpointManager.GetInstanceStateResponse) response).getCheckpoint());
+  }
+
+  @Test
+  public void testCleanStatefulCheckpoint() throws Exception {
+    prepareClient(SimpleCheckpointManagerClient.RequestType.CLEAN_STATEFUL_CHECKOINTS);
+
+    runClient(simpleCheckpointManagerClient);
+    Thread.sleep(2 * 1000);
+
+    verify(backendStorage).dispose(anyString(), anyString(), anyBoolean());
+
+    Message response = simpleCheckpointManagerClient.getResponse();
+    Assert.assertTrue(response instanceof CheckpointManager.CleanStatefulCheckpointResponse);
```

Comment:
Let's use static imports for all the `Assert.assertFoo` calls as well to improve readability.